lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -105,7 +105,7 @@ def get(self):
                     host_value = header
         if not origin:
             origin = host_value
-        host = f"http://{origin}"
+        host = f"{request.headers.get('X-Forwarded-Proto')}://{origin}"
```

Comment:
Should we instead have something like `proto = request.headers.get("X-Forwarded-Proto", "https")` to account for the possibility of an external NGINX that *isn't* configured to pass `X-Forwarded-Proto` by defaulting???

I think it's great to have NGINX confirm the protocol it's using, but I think we want to assume/require `https` so we might as well make that explicit.

And if we decide that we want to *require* the header, then we need to fail startup without it since `/api/v1/endpoints` won't deliver usable URIs!

Reply:
> Should we instead have something like `proto = request.headers.get("X-Forwarded-Proto", "https")` to account for the possibility of an external NGINX that _isn't_ configured to pass `X-Forwarded-Proto` by defaulting???

Thats a good point. Yeah, we should account for the possibility of an external NGINX not passing the right headers.

Reply:
I think we should either require the header (and fail bitterly if it's missing) or be prepared to operate without it (presumably by demanding that it be `https` if it's not specified).  The problem is, neither of those produces a great experience in the failure cases:  in the first scenario, everyone wonders why it doesn't work until someone realizes the problem is in the proxy, while in the second the Dashboard doesn't work (which looks bad for us) because the server tells it to use `https` when that connection doesn't necessarily work.

I really don't know what we should do if we can't tell what schema the client is using...but, failing hard and early is usually a good plan, especially if we can make it plain why we're failing.